### PR TITLE
Add hash capabilities to OUI

### LINKED ISSUE #224: OUI hashability
# Summary
`OUI` does not have a `__hash__` function.  This prevents the use of `OUI` for keys.
- Python Hash documentation [link](https://docs.python.org/3/reference/datamodel.html#object.__hash__)

# Example - Break
```
from netaddr import OUI

a = OUI("002272")

try:
    set(a)
except TypeError:
    print("Hash failed")
```

# Example - Fix
```
from netaddr import OUI

a = OUI("002272")
set(a)
```

### PR DESCRIPTION
**Pull request by amitmi704**
Closes #224

### CODE REVIEW

File: netaddr/eui/__init__.py [hunk 1]
```diff
@@ -100,6 +100,10 @@ def __init__(self, oui):
         else:
             raise NotRegisteredError('OUI %r not registered!' % (oui,))
 
+    def __hash__(self):
+        """:return: hash of this OUI object suitable for dict keys, sets etc"""
```
Comment (jstasiak) [inline netaddr/eui/__init__.py:104]: Let's drop this comment, the meaning of the value returned by `__hash__()` is rather standard.

**Review by jstasiak — CHANGES_REQUESTED** [review]
Hey, thanks for the patch. Can you provide a unit test for this?

File: netaddr/eui/__init__.py (hunk 1 above)
Comment (amitmi704) [reply netaddr/eui/__init__.py:104]: Will do.  That was a copy-and-paste from the `EUI.__hash__`.

File: netaddr/tests/eui/test_eui.py [hunk 2]
```diff
@@ -158,6 +158,7 @@ def test_eui_oui_information():
 def test_oui_constructor():
     oui = OUI(524336)
     assert str(oui) == '08-00-30'
+    assert hash(oui) == 524336
```
Comment (jstasiak) [inline netaddr/tests/eui/test_eui.py:161]: We need a real-world hash-capability demonstration here. If the original issue is about using `OUI` objects as dictionary keys let's use that to check if it works, we shouldn't be hardcoding specific hash values (those can't be depended on etc.).

Comment (amitmi704) [reply netaddr/tests/eui/test_eui.py]: Fair enough, let me know if you want me to drop the `hash` asserts in the new one.  Broke it out into a different func to not clutter the `test_oui_constructor`.

Comment (jstasiak) [reply netaddr/tests/eui/test_eui.py]: Thank you! Yes, please drop those two asserts and I'm happy to merge this.

**Comment by jstasiak** [comment]
Thank you!

**Pull request merged**